test/Element/MonthTest.php

```diff
@@ -81,4 +82,11 @@ public function testHTML5MonthValidation(string $value, bool $expected): void
         $monthValidator = $inputSpec['validators'][0];
         self::assertEquals($expected, $monthValidator->isValid($value));
     }
+
+    public function testTestDayTimeReset(): void
+    {
+        $element = new MonthElement('foo');
+        $date    = DateTime::createFromFormat($element->getFormat(), '2023-01');
```

Comment:
@samuelgmartins this test is fine, but is there anything from `$element` that we can use to fetch a generated date, perhaps? :thinking:

Reply:
I don't think so as all we know about the value is the year and month. I think I would prefer to see a check for the 31st/30th if those are the erroneous dates rather than enforcing date 01.

Reply:
```php
$date    = DateTime::createFromFormat($element->getFormat(), date('Y-m'));
```
That way?

Reply:
I meant manually setting the date to the 31st of a month and then checking if the original issue still exists.

Reply:
@internalsystemerror I think it's not possible because you need to change the date on the server to the 31st.

I tried to set it directly in the DataStep object using the setBaseValue() function, but this is linked to the format, I tried to send the dataTime object already created, even though I couldn't do the test to check exactly the 31st. 
Follows failed test:
```php
use Laminas\Form\Form;
use Laminas\Form\Element;
use Laminas\InputFilter\InputFilter;
use Laminas\InputFilter\Input;
use Laminas\Validator\DateStep;

$form  = new Form();
$element = new Element\Month('month');
$input = new Input();
$input
    ->getValidatorChain()
    ->attach((new DateStep())
        ->setFormat($element->getFormat())
        ->setStep(new \DateInterval("P1M"))
        ->setBaseValue(\DateTime::createFromFormat('Y-m-d', '1970-01-31', new \DateTimeZone('UTC')))
    )
;
$inputFilter = new InputFilter();
$inputFilter->add($input, 'month');
$form->setUseInputFilterDefaults(false);
$form->setInputFilter($inputFilter);
$form->add($element);
$form->setData(['month' => '2000-01']); // 31 day month
var_dump($form->isValid()); // returns false
var_dump($form->getMessages()); // dateStepNotStep
$form->setData(['month' => '2000-04']); // 30 day month
var_dump($form->isValid()); // returns false
var_dump($form->getMessages()); // dateStepNotStep
```
So, as suggested, I ended up checking to see if the date was reset.

Reply:
Yeah I think the problem stems from this part of the DateStep validator https://github.com/laminas/laminas-validator/blob/c5f73b1dc9b657af24736b56318bcd880d9fee94/src/DateStep.php#L254. We could probably use a psr/clock implementation there, but that shouldn't hold this PR up.

Reply:
I wonder now though, if that is a bug with the validator?

Reply:
> I wonder now though, if that is a bug with the validator?

I believe not because the DateStep validator has to validate other elements with full date and time, being an individuality relative to the Month element that validates only the year and month.